opentelekomcloud/resource_opentelekomcloud_obs_bucket.go

```diff
@@ -248,15 +251,71 @@ func resourceObsBucket() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"credentials": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"access_key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"secret_key": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
+func obsClient(d *schema.ResourceData, config *Config) (*obs.ObsClient, error) {
+	errMsg := "error creating OpenTelekomCloud OBS client: %s"
+
+	credentials, ok := d.GetOk("credentials")
+	if !ok {
+		client, err := config.newObjectStorageClient(GetRegion(d, config))
+		if err != nil {
+			err = fmt.Errorf(errMsg, err)
+		}
+		return client, err
+	}
+
+	// if credentials are set in the resource, mimics config.newObjectStorageClient
+
+	c, err := openstack.NewOBSService(config.HwClient, golangsdk.EndpointOpts{
+		Region:       GetRegion(d, config),
+		Availability: config.getHwEndpointType(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// init log
+	setUpOBSLogging()
+
+	credentialsSlice := credentials.([]interface{})
+	if len(credentialsSlice) == 0 {
+		return nil, fmt.Errorf("credentials are missing")
+	}
+	credMap := credentialsSlice[0].(map[string]interface{})
+	client, err := obs.New(credMap["access_key"].(string), credMap["secret_key"].(string), c.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf(errMsg, err)
+	}
+	return client, nil
+}
+
 func resourceObsBucketCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	obsClient, err := config.newObjectStorageClient(GetRegion(d, config))
+	client, err := obsClient(d, config)
```

Comment:
now it logs in obs, but it could be surprise for a user

Reply:
`obsClient` wraps `config.newObjectStorageClient(GetRegion(d, config))`. When using config without `credentials`, behaviour is not changing.